docs/calendar-draft.md

```diff
@@ -0,0 +1,315 @@
+	[Temporal.Calendar.toISO] = (self) => {
+		return self;
+	},
+
+	[Temporal.Calendar.fromISO] = (isoDate) => {
+		return isoDate;
+	},
+
+	[Temporal.Calendar.id] = "iso",
+
+	// ALL OTHER METHODS:
+	[Temporal.Calendar.plus] = () => {
+		throw new TypeError("Unsupported operation: full calendar required");
+	}
+	// Same for [Temporal.Calendar.minus], etc.
+}
+```
+
+It would in effect render default Temporal.Date (and Temporal.DateTime) with fewer operations until you specify a calendar.  The following methods/getters would throw:
+
+- .dayOfWeek
+- .weekOfYear
+- .daysInMonth
+- .daysInYear
+- .leapYear
+- .plus() -- *might* be OK if the programmer requests only time units
+- .minus() -- *might* be OK if the programmer requests only time units
+- .difference() -- *might* be OK if the programmer requests only time units
+- .getYearMonth()
+- .getMonthDay()
+
+The following methods/getters would still work:
+
+- .with()
+- .withTime()
+- .toString()
+- .toLocaleString()
+- .compare()
+
+Although small, this set of operations still covers many of the recipes in the Temporal Cookbook.
+
+To enable the extended set of operations, the user would just use `.withCalendar()`:
+
+```javascript
+// Force the Gregorian calendar:
+Temporal.Date.from("2019-12-06").withCalendar("gregory").weekOfYear;
+
+// Use a calendar from another source:
+Temporal.Date.from("2019-12-06").withCalendar(Intl.defaultCalendar).weekOfYear;
+Temporal.Date.from("2019-12-06").withCalendar(request.calendar).weekOfYear;
+```
+
+The Gregorian calendar, ID "gregory" ([why not "gregorian"?](https://github.com/tc39/ecma402/issues/212)), could be used to indicate the full ISO calendar.  So, "iso" would mean the partial version, and "gregory" would mean the full version.  Alternatively, "iso8601", the identifier defined by CLDR as "Gregorian calendar using the ISO 8601 calendar week rules", could be the identifier for the full ISO calendar.
+
+## Temporal.Date API changes
+
+### New Temporal.Date instance methods
+
+The following methods involving ISO conversion would be added:
+
+```javascript
+Temporal.Date.prototype.toISO = function(): Temporal.Date {
+	const isoDate = this.calendar[Temporal.Calendar.toISO](this);
+	// assert: isoDate.calendar === Temporal.Calendar.iso
+	return isoDate;
+}
+
+// Temporal.Date.prototype.with does *not* modify the calendar. A new method
+// is added for that:
+
+Temporal.Date.prototype.withCalendar = function(newCalendar: Calendar) {
+	const isoDate = this.toISO();  // note: call intrinsic version
+	const otherDate = newCalendar[Temporal.Calendar.fromISO](isoDate);
+	// assert: otherDate.calendar === newCalendar
+	return otherDate;
+}
+```
+
+### ISO strings with calendar hint
+
+Serialization to/from strings will still be supported.  To convert to a string, the `Temporal.Calendar.id` field of the calendar will be appended in `[]` following the date.  For example, `2019-12-06[hebrew]` refers to 2019-12-06 projected into the Hebrew calendar.
```

Comment:
How will this compose with time zone indicators for Temporal.DateTime instances? `2019-12-06[hebrew]T16:28[America/New_York]`?

Reply:
No!

Calendars apply to ALL temporal objects including Time! Think a martian calendar where days don’t have 24h

so it would be 2019-12-06T16:23+00:50[America/NewYork][hebrew]

Reply:
Which is ok, since IANA zones all have a / and calendar names shall not

Reply:
There are **many** aliases without `/`, at least one of which you personally considered canonical in the recent past: https://github.com/tc39/proposal-temporal/issues/156#issuecomment-531039683

The multiple-bracketed-suffixes format is ambiguous and should not be used.

Reply:
I wonder if it would help with disambiguation to do `[c=hebrew]` instead.  Then we could also support adding more such options in the future.  Would we want to change time zone to `[z=America/NewYork]` also?

Reply:
New thread: #293